python/plugins/processing/algs/qgis/SpatialiteExecuteSQL.py

```diff
@@ -56,16 +56,19 @@ def displayName(self):
         return self.tr('Spatialite execute SQL')
 
     def processAlgorithm(self, parameters, context, feedback):
-        database = self.getParameterValue(self.DATABASE)
-        uri = QgsDataSourceUri(database)
+        database = self.parameterAsVectorLayer(parameters, self.DATABASE, context)
+        databaseuri = database.dataProvider().dataSourceUri()
+        uri = QgsDataSourceUri(databaseuri)
         if uri.database() is '':
-            if '|layerid' in database:
-                database = database[:database.find('|layerid')]
-            uri = QgsDataSourceUri('dbname=\'%s\'' % (database))
+            if '|layerid' in databaseuri:
+                databaseuri = databaseuri[:databaseuri.find('|layerid')]
+            uri = QgsDataSourceUri('dbname=\'%s\'' % (databaseuri))
         self.db = spatialite.GeoDB(uri)
```

Comment:
This should be just 'db', not self.db. In the new api algorithm execution should not alter the state of the algorithm instance itself.